x-pack/plugins/reporting/server/routes/diagnostic/screenshot.ts

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { ReportingCore } from '../..';
+import { API_DIAGNOSE_URL } from '../../../common/constants';
+import { omitBlacklistedHeaders } from '../../export_types/common';
+import { getAbsoluteUrlFactory } from '../../export_types/common/get_absolute_url';
+import { generatePngObservableFactory } from '../../export_types/png/lib/generate_png';
+import { LevelLogger as Logger } from '../../lib';
+import { DiagnosticResponse } from '../../types';
+import { authorizedUserPreRoutingFactory } from '../lib/authorized_user_pre_routing';
+
+export const registerDiagnoseScreenshot = (reporting: ReportingCore, logger: Logger) => {
+  const setupDeps = reporting.getPluginSetupDeps();
+  const userHandler = authorizedUserPreRoutingFactory(reporting);
+  const { router } = setupDeps;
+
+  router.post(
+    {
+      path: `${API_DIAGNOSE_URL}/screenshot`,
+      validate: {},
+    },
+    userHandler(async (user, context, req, res) => {
+      const generatePngObservable = await generatePngObservableFactory(reporting);
+      const config = reporting.getConfig();
+      const decryptedHeaders = req.headers as Record<string, string>;
+      const [basePath, protocol, hostname, port] = [
+        config.kbnConfig.get('server', 'basePath'),
+        config.get('kibanaServer', 'protocol'),
+        config.get('kibanaServer', 'hostname'),
+        config.get('kibanaServer', 'port'),
+      ] as string[];
+
+      const getAbsoluteUrl = getAbsoluteUrlFactory({
```

Comment:
it is so weird that this utility has a factory function

Reply:
Yeah, I don't really think it's useful to have it as a factory since it's really just closuring the basepath and a few other URL parameters. I think we can clean it up, but don't want it here since it's expanding the scope of the PR